core/templates/dev/head/profile/PreferencesSpec.js

```diff
@@ -0,0 +1,58 @@
+// Copyright 2014 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+describe('Preferences Controller', function() {
+  describe('PreferencesCtrl', function() {
+    var scope, ctrl, $Http;
+
+    beforeEach(function() {
+      module('oppia');
+    });
+
+    beforeEach(inject(function(_$httpBackend_, $http, $rootScope, $controller) {
+      _$httpBackend_.expectGET('/signuphandler/data').respond({
```

Comment:
Yep, your first paragraph is right!

Re not implementing the second test -- yes, that is fine. Just one test will do; I don't think the second one makes much sense as a frontend one anyway since the whole point there is to test server state. An e2e test would be better.